aurelia-slickgrid/src/aurelia-slickgrid/aurelia-slickgrid.ts

```diff
@@ -659,19 +658,19 @@ export class AureliaSlickgridCustomElement {
       // wait for the "collectionAsync", once resolved we will save it into the "collection"
       // the collectionAsync can be of 3 types HttpClient, HttpFetch or a Promise
       //
-      collectionAsync.then((response: HttpResponseMessage | Response | any[]) => {
-        if (response instanceof Response && typeof response.json === 'function') {
+      collectionAsync.then((response: any | any[]) => {
+        if (Array.isArray(response)) {
+          this.updateEditorCollection(column, response); // from Promise
+        } else if (response instanceof Response && typeof response.json === 'function') {
           if (response.bodyUsed) {
             throw new Error('[Aurelia-SlickGrid] The response body passed to collectionAsync was ' +
               'already read. Either pass the dataset from the Response ' +
               'or clone the response first using response.clone()');
           }
-
+          // from aurelia-fetch-client
```

Comment:
might be confusing to have this comment here. A Promise may or may not come from aurelia-fetch-client

Reply:
I've put the comments to know where each came from. It was a bit confusing to know the source of each. `aurelia-http-client` and `aurelia-fetch-client`, are not using the same and regular `Promise` are catch by the 1st `if`. Do you have a better way of identifying this?

Reply:
The `Response` interface is not specific to `aurelia-fetch-client`. For example, your consumer could be using the native `fetch` api and you would still get the `Response` interface. Not sure you need to identify which API the user is using. I think that is part of the beauty of getting the `Response` object back.

Reply:
@jmzagorski 
are you suggesting to remove the comment completely? I just find it a little less clear doing so, but if that removes confusion then sure

Reply:
you can keep it if it seems more clear, I have no problem with that. However, be aware that I can use the https://github.com/github/fetch library and it will return `Response` too

Reply:
I'll change it for `// from Fetch` then :)